components/script/script_module.rs

```diff
@@ -162,8 +164,15 @@ pub struct ModuleTree {
     text: DomRefCell<DOMString>,
     record: DomRefCell<Option<ModuleObject>>,
     status: DomRefCell<ModuleStatus>,
-    parent_urls: DomRefCell<HashSet<ServoUrl>>,
-    descendant_urls: DomRefCell<HashSet<ServoUrl>>,
+    // The spec maintains load order for descendants,
+    // so we use an indexset for descendants and parents.
+    // This isn't actually necessary for parents however
+    // the IndexSet APIs don't interop with HashSet, and
+    // IndexSet isn't very expensive (https://github.com/bluss/indexmap/issues/110)
+    //
+    // https://html.spec.whatwg.org/multipage/webappapis.html#fetch-the-descendants-of-a-module-script
```

Comment:
nit: `https://html.spec.whatwg.org/multipage/#fetch-the-descendants-of-a-module-script`